libs/ui-lib/lib/common/components/ui/formik/UploadField.tsx

```diff
@@ -36,64 +41,77 @@ const UploadField: React.FC<UploadFieldProps> = ({
+        onFileInputChange={(_event, file) => {
+          setFilename(file.name);
+          setTouched(true);
+          setValue(file);
+        }}
+        onBlur={(e) => {
+          const file: File | string = field.value;
+          if (file instanceof File) {
+            const reader = new FileReader();
 
-  const fieldHelperText = <HelperText fieldId={fieldId}>{helperText}</HelperText>;
+            reader.onload = (event) => {
+              if (event.target) {
+                const contentAsString = event.target.result as string;
+                setValue(trimSshPublicKey(contentAsString));
+                onBlur && onBlur(e);
+              }
+            };
 
-  return (
-    <Stack>
-      <StackItem>
-        <FormGroup
-          fieldId={fieldId}
-          label={label}
-          helperText={fieldHelperText}
-          helperTextInvalid={fieldHelperText}
-          validated={isValid ? 'default' : 'error'}
-          isRequired={isRequired}
-          labelIcon={labelIcon}
-        >
-          {children}
-          <FileUpload
-            filenamePlaceholder={t('ai:Drag a file here or browse to upload')}
-            browseButtonText={t('ai:Browse...')}
-            clearButtonText={t('ai:Clear')}
-            id={field.name}
-            style={{ resize: 'vertical' }}
-            validated={isValid ? 'default' : 'error'}
-            isRequired={isRequired}
-            aria-describedby={`${fieldId}-helper`}
-            type="text"
-            value={field.value as string}
-            filename={filename}
-            onChange={(value, filename) => {
-              setFilename(filename);
-              helpers.setTouched(true);
-              helpers.setValue(value);
-            }}
-            onBlur={(e) => {
-              field.onBlur(e);
-              onBlur && onBlur(e);
-            }}
-            onReadStarted={() => setIsFileUploading(true)}
-            onReadFinished={() => setIsFileUploading(false)}
-            isLoading={isFileUploading}
-            disabled={isDisabled}
-            dropzoneProps={{
-              ...dropzoneProps,
-              onDropRejected:
-                dropzoneProps?.onDropRejected && dropzoneProps?.onDropRejected(helpers),
-            }}
-            allowEditingUploadedText={allowEdittingUploadedText}
-          />
-        </FormGroup>
-      </StackItem>
-      <StackItem>
-        {errorMessage && (
-          <HelperText fieldId={fieldId} isError>
-            {errorMessage}
+            // Read the file content as text
+            reader.readAsText(file);
+          } else {
+            setValue(trimSshPublicKey(file));
+            onBlur && onBlur(e);
+          }
+        }}
+        onReadStarted={() => setIsFileUploading(true)}
+        onReadFinished={() => setIsFileUploading(false)}
+        isLoading={isFileUploading}
+        disabled={isDisabled}
+        allowEditingUploadedText={allowEdittingUploadedText}
+        dropzoneProps={{
```

Comment:
I see now, regarding my previous comment in `UploadSSH`.

Since `UploadField` is a generic component, it shouldn't declare its own constants for the accepted file types, max accepted file size, etc.

IMO we should keep the restrictions in `UploadSSH` and pass them to this component.

Reply:
The question is that we use UploadField only inside UploadSSH. Maybe is better to have only one component?

Reply:
I still think it's a good idea to have a generic, reusable component. 

I think the main problem may be about the `trimSssh` part. If it can be put back in `UploadSSH`, then I'd rather keep the 2 components. Whatever you feel is better.

Reply:
Ok. I keep UploadField as a generic component.